src/v/raft/types.h

```diff
@@ -309,6 +309,32 @@ struct snapshot_metadata {
     ss::lowres_clock::time_point cluster_time;
 };
 
+struct producer_identity {
+    int64_t id;
+    int16_t epoch;
+
+    bool operator==(const producer_identity& other) const {
+        return id == other.id && epoch == other.epoch;
+    }
+
+    bool operator!=(const producer_identity& other) const {
```

Comment:
i think this method is also compiler generated. 

if you need this i'd suggest using the new `spaceship` operator `<=>` to define them all in one method.

Reply:
I think you do get `!=` for free when you default create `==`. don't quote me on that